storage/google/cloud/storage/blob.py

```diff
@@ -348,28 +370,23 @@ def download_to_file(self, file_obj, client=None):
         :raises: :class:`google.cloud.exceptions.NotFound`
         """
         client = self._require_client(client)
-        if self.media_link is None:  # not yet loaded
-            self.reload()
-
-        download_url = self.media_link
-
-        # Use apitools 'Download' facility.
-        download = Download.from_stream(file_obj)
-
-        if self.chunk_size is not None:
-            download.chunksize = self.chunk_size
-
+        # Get the download URL.
+        download_url = self._get_download_url()
+        # Get any extra headers for the request.
         headers = _get_encryption_headers(self._encryption_key)
-
-        request = Request(download_url, 'GET', headers)
-
-        # Use ``_base_connection`` rather ``_connection`` since the current
-        # connection may be a batch. A batch wraps a client's connection,
-        # but does not store the ``http`` object. The rest (API_BASE_URL and
-        # build_api_url) are also defined on the Batch class, but we just
-        # use the wrapped connection since it has all three (http,
-        # API_BASE_URL and build_api_url).
-        download.initialize_download(request, client._base_connection.http)
+        # Create a ``requests`` transport with the client's credentials.
+        transport = tr_requests.AuthorizedSession(client._credentials)
+
+        # Download the content.
+        if self.chunk_size is None:
+            download = resumable_media.Download(download_url, headers=headers)
+            response = download.consume(transport)
+            file_obj.write(response.content)
```

Comment:
@tseaver The interface for `Download` intentionally **does not** take a `stream` / `file`-like input. And even if it did, it would use `stream.write(response.content)` internally, so it doesn't save any RAM.